pkg/types/validation/installconfig.go

```diff
@@ -609,3 +611,29 @@ func validateIPProxy(proxy string, n *types.Networking, fldPath *field.Path) fie
 	}
 	return allErrs
 }
+
+// validateWorkloadPartition ensures that the given list of workload partitions either empty or:
+//  - The name of the partition must be "management" (for the first release)
+//  - Made of unique names (no duplicates)
+//  - With a valid-looking CPU set description (note: Cannot actually verify against real hardware)
+func validateWorkload(partitions []types.WorkloadPartition, fldPath *field.Path) field.ErrorList {
+	partitionNames := map[types.WorkloadPartitionName]bool{}
+	allErrs := field.ErrorList{}
+	for i, p := range partitions {
+		partitionFldPath := fldPath.Index(i)
+		if p.Name != types.ManagementWorkloadPartition {
+			allErrs = append(allErrs, field.NotSupported(partitionFldPath.Child("name"), p.Name, []string{string(types.ManagementWorkloadPartition)}))
+		}
+		if partitionNames[p.Name] {
+			allErrs = append(allErrs, field.Duplicate(partitionFldPath.Child("name"), p.Name))
+		}
+		partitionNames[p.Name] = true
+		if p.CPUIds == "" {
+			allErrs = append(allErrs, field.Invalid(partitionFldPath.Child("cpuIDs"), p.CPUIds, "cannot be empty"))
+		} else if cpuset, err := cpuset.Parse(p.CPUIds); err != nil || cpuset.IsEmpty() {
+			// Re-check if the parsed cpuset is empty, since the parser may interpret some errors as the empty set.
+			allErrs = append(allErrs, field.Invalid(partitionFldPath.Child("cpuIDs"), p.CPUIds, "could not parse the cpuset"))
+		}
```

Comment:
Do we need to check for empty twice? If the `IsEmpty` check sufficient?
```suggestion
		swtich cpuset, err := cpuset.Parse(p.CPUIds); {
		case err != nil:
			// invalid error
		case cpuset.IsEmpty():
			// required error
		}
```